packages/upload-client/src/index.js

```diff
@@ -183,5 +195,24 @@ async function uploadBlockStream(
   if (!root) throw new Error('missing root CID')
 
   await Upload.add(conf, root, shards, options)
```

Comment:
Any reason to not `Upload.add` as last? If something fails, I would prefer that `Upload` was not recorded...

Reply:
Have moved to last.